packages/database/src/core/modelClasses/Entity.js

```diff
@@ -538,27 +538,55 @@ export class EntityModel extends MaterializedViewLogDatabaseModel {
 
   async buildSyncLookupQueryDetails() {
     return {
+      // TODO: Remove survey response entities and task entities
+      // when MAUI-5722 is complete
       ctes: [
         `
-          entity_parent_child_relation_union AS (
-            SELECT parent_id as entity_id, entity_hierarchy_id 
-            FROM entity_parent_child_relation
+          entities_to_sync AS (
+            -- root project entities
+            SELECT entity.id as entity_id, project.entity_hierarchy_id
+            FROM entity join project on entity.id = project.entity_id
+
             UNION
+
+            -- all child entities of root project entities
             SELECT child_id as entity_id, entity_hierarchy_id 
             FROM entity_parent_child_relation
+
+            UNION
+
+            -- survey response entities
+            SELECT survey_response.entity_id, project.entity_hierarchy_id
+            FROM survey_response
+            JOIN survey ON survey.id = survey_response.survey_id
+            JOIN project ON project.id = survey.project_id
+
+            UNION
+
+            -- task entities
+            SELECT task.entity_id, project.entity_hierarchy_id
+            FROM task
+            JOIN survey ON survey.id = task.survey_id
+            JOIN project ON project.id = survey.project_id
```

Comment:
You're right this isn't thaaaat bad...
I do like to avoid defensive programming though. Prefer to take this out now and then put back in if really needed rather than put in now and let ourselves get away with bad data